File: lection04/app_java/src/main/java/ru/hse/lection04/presentationlayer/adapter/LogViewHolder.java
package ru.hse.lection04.presentationlayer.adapter;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import ru.hse.lection04.R;
import ru.hse.lection04.objects.LogEntry;

public class LogViewHolder extends RecyclerView.ViewHolder {
    protected static final String MESSAGE_PATTERN = "[%s] - %s";


    protected final TextView mText;


    public LogViewHolder(@NonNull View itemView) {
        super(itemView);

        mText = itemView.findViewById(R.id.text);
    }


    public void bind(LogEntry item) {
        final String message = String.format(MESSAGE_PATTERN, item.time, item.message);
        mText.setText(message);
    }
}

File: lection04/app_kotlin/src/main/java/ru/hse/lection04/presentationlayer/adapter/LogViewHolder.kt
package ru.hse.lection04.presentationlayer.adapter

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ru.hse.lection04.R
import ru.hse.lection04.objects.LogEntry

class LogViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    companion object {
        protected const val MESSAGE_PATTERN = "[%s] - %s"
    }


    protected val text: TextView


    init {
        text = itemView.findViewById(R.id.text)
    }


    fun bind(item: LogEntry?) {
        text.text = String.format(MESSAGE_PATTERN, item?.time, item?.message)
    }
}